packages/@aws-cdk/aws-iam/lib/policy-document.ts

```diff
@@ -90,12 +91,16 @@ export class AccountPrincipal extends ArnPrincipal {
  * An IAM principal that represents an AWS service (i.e. sqs.amazonaws.com).
  */
 export class ServicePrincipal extends PolicyPrincipal {
-  constructor(public readonly service: string) {
+  constructor(public readonly service: string, private readonly region?: string) {
     super();
   }
 
   public policyFragment(): PrincipalPolicyFragment {
-    return new PrincipalPolicyFragment({ Service: [ this.service ] });
+    const factName = Facts.servicePrincipal(this.service);
+    const principal = this.region != null
```

Comment:
Why the two operating modes?

Reply:
One uses the current stack's region, the other uses a user-specific region (assuming, for example, we have to trust a service from a different region).

Reply:
Oh I see. Got it.

Reply:
I think the tokenization should happen here and not in the region-info library.

Reply:
@eladb  that means _every_ usage site that requires a form of tokenization will have to make up their own `Token` subclass. I don't mind - it just feels... heavy for a thing that can be done once and for all.